File: 3329-find-the-length-of-the-longest-common-prefix/3329-find-the-length-of-the-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, arr1, arr2):
        """
        :type arr1: List[int]
        :type arr2: List[int]
        :rtype: int
        """
        prefixes = set()
        
        for val in arr1:
            while val not in prefixes and val > 0:
                prefixes.add(val)
                val /= 10
        
        longestPrefixCount = 0

        for val in arr2:
            while val not in prefixes and val > 0:
                val /= 10
            if val > 0:
                string = str(val)
                longestPrefixCount = max(longestPrefixCount, len(string))
        

        return longestPrefixCount
        

File: 3329-find-the-length-of-the-longest-common-prefix/3329-find-the-length-of-the-longest-common-prefix.cpp
class Solution {
public:
    int longestCommonPrefix(vector<int>& arr1, vector<int>& arr2) {
        unordered_set<int> arr1Prefixes;
        for(int val: arr1){
            while(!arr1Prefixes.count(val) && val > 0){
                arr1Prefixes.insert(val);
                val /= 10;
            }
        }
        
        int longestPrefix = 0;

        for(int val: arr2){
            while(!arr1Prefixes.count(val) && val > 0){
                val /= 10;
            }
            if(val > 0){
                string str = to_string(val);
                // longestPrefix = max(longestPrefix, str.length());
                if (longestPrefix < str.length()){
                    longestPrefix = str.length();
                }
            }
        }
        return longestPrefix;
    }
};